packages/tpetra/core/example/Finite-Element-Assembly/typedefs.hpp

```diff
@@ -50,13 +50,12 @@
 #include <Tpetra_CrsGraph.hpp>
 #include <Tpetra_CrsMatrix.hpp>
 
+typedef double Scalar;
+
 // Get LocalOrdinal & GlobalOrdinal from Map defaults.
 typedef Tpetra::Map<>::local_ordinal_type LocalOrdinal;
 typedef Tpetra::Map<>::global_ordinal_type GlobalOrdinal;
-
-typedef double Scalar;
-
-typedef Tpetra::Details::DefaultTypes::node_type NT;
+typedef Tpetra::Map<>::node_type NT;
 
 typedef Kokkos::DefaultExecutionSpace ExecutionSpace;
```

Comment:
Regarding line 61: TL;DR is "do this: `typedef Tpetra::Map<>::execution_space ExecutionSpace;"

Explanation: Tpetra's default execution space (`Tpetra::Map<>::node_type::execution_space`, same as `Tpetra::Map<>::execution_space`) may not necessarily be the same as Kokkos' default execution space (`Kokkos::DefaultExecutionSpace`).  This is beacuse Tpetra may use _any_ enabled Kokkos execution space as Tpetra's default.  For example, some users want to use Kokkos threads, but don't want Tpetra and downstream solvers to use threads by default.  (See also #1939 and its underlying justification.)

(btw my view is that telling Tpetra and downstream solvers not to use threads, should happen entirely as a run-time option.  It's bonkers that Trilinos has so many configuration options.  If you turn on OpenMP in the build, you should get OpenMP.  If you built with a CUDA compiler, you should get CUDA.)

It's OK to use Kokkos' default execution space, even if it differs from Tpetra's default execution space, as long as Kokkos has been initialized.  (`Kokkos::initialize` always initializes Kokkos' default execution space.  `Tpetra::Map`' constructor always makes sure that the Map's execution space has been initialized.)  However, it is important to get the memory spaces right.  What if Kokkos uses CUDA by default, but Tpetra does not?  You'll have troubles if you launch CUDA kernels with Kokkos that attempt to access Tpetra's host memory.